### PR TITLE
Feature/infinite scroll in newsfeedpage bundle

### PR DESCRIPTION
**Pull request by ashgaliyev**
Merged infinite scroll changes to new NewsfeedPage bundle

### CODE REVIEW

File: app/javascript/App/bundles/NewsfeedPage/NewsList.tsx [hunk 1]
```diff
@@ -58,30 +58,20 @@ class NewsList extends React.Component<Props, State> {
   }
 
   closeTips() {
-    this.props.newsfeedTips()
+    this.props.toggleNewsfeedTips()
```
Comment (alexfedoseev) [inline app/javascript/App/bundles/NewsfeedPage/NewsList.tsx]: These two names are misleading: toggle or close?

File: app/javascript/App/bundles/NewsfeedPage/NewsList.tsx [hunk 2]
```diff
@@ -178,7 +149,7 @@ class NewsList extends React.Component<Props, State> {
             : {}
         }
       >
-        {this.renderView() /* viewState, initialRenderTips, readNewsIds, isLoading, fetchMoreNewsFeed */}
+        {this.renderView()}
```
Comment (alexfedoseev) [inline app/javascript/App/bundles/NewsfeedPage/NewsList.tsx]: Just FYI: usually `renderSomething()` is a sign that you need new component, but not critical.

File: app/javascript/App/bundles/NewsfeedPage/NewsfeedPage.tsx [hunk 3]
```diff
@@ -25,20 +25,22 @@ import { CoinList, Coin } from '../common/types';
 
 const STATUSES = {
   LOADING: 'Loading',
+  INFINITE_SCROLL_LOADING: 'InfiniteScrollLoading',
```
Comment (alexfedoseev) [inline app/javascript/App/bundles/NewsfeedPage/NewsfeedPage.tsx]: FWIW I usually mirror values, e.g. `INFINITE_SCROLL_LOADING: 'INFINITE_SCROLL_LOADING'` but not critical as well.

File: app/javascript/App/bundles/NewsfeedPage/NewsfeedPage.tsx [hunk 4]
```diff
@@ -63,11 +65,16 @@ class NewsfeedPage extends React.Component<Props, State> {
   }
 
   componentDidMount() {
+    this.setState({
+      status: STATUSES.LOADING
```
Comment (alexfedoseev) [inline app/javascript/App/bundles/NewsfeedPage/NewsfeedPage.tsx]: Why not setting this in initial state and exec callback here w/o flipping the state?

File: app/javascript/App/bundles/NewsfeedPage/NewsfeedPage.tsx [hunk 5]
```diff
@@ -122,11 +133,25 @@ class NewsfeedPage extends React.Component<Props, State> {
...
+    this.setState({
+      status: STATUSES.INFINITE_SCROLL_LOADING
+    });
+
+    localAPI.get(`/news`, { publishedUntil: lastNews.publishedUntil }).then(response => {
+        this.setState({
+          status: STATUSES.READY,
+          sortedNewsItems: [...this.state.sortedNewsItems, ...response.payload.sort(this.sortNewsFunc)]
+        })
+    });
```
Comment (alexfedoseev) [inline app/javascript/App/bundles/NewsfeedPage/NewsfeedPage.tsx]: You should trigger this from within callback of `setState`.

File: app/javascript/App/bundles/NewsfeedPage/NewsfeedPage.tsx [hunk 6]
```diff
@@ -176,42 +201,32 @@ class NewsfeedPage extends React.Component<Props, State> {
...
+                <NewsListHeader
+                  coins={this.props.coinList}
+                  feedSources={this.props.feedSources}
+                  showFilters={this.state.showFilters}
+                  activeFilters={this.state.activeFilters}
+                  newsfeedTips={this.state.newsfeedTips}
+                />
+                <NewsList
+                  isLoading={() => this.state.status === STATUSES.LOADING}
+                  isInfiniteScrollLoading={() => this.state.status === STATUSES.INFINITE_SCROLL_LOADING}
```
Comment (alexfedoseev) [inline app/javascript/App/bundles/NewsfeedPage/NewsfeedPage.tsx]: It prolly can be just booleans.

File: app/javascript/App/bundles/common/components/CoinTags.tsx [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+
+interface Props {
+  itemWithCoinLinkData: ItemWithCoinLinkData,
+  selectCoin: (CoinLinkData) => void,
+}
+
+const CoinTags = ({ itemWithCoinLinkData, selectCoin }: Props) => (
+  <div>
+    {itemWithCoinLinkData.coin_link_data.map((data, index) => (
+      <a key={index} className="tag pointer" onClick={() => selectCoin(data)}>
```
Comment (alexfedoseev) [inline app/javascript/App/bundles/common/components/CoinTags.tsx]: Note on index-as-key.

File: app/javascript/App/bundles/NewsfeedPage/NewsList.tsx (hunk 1 above)
Comment (ashgaliyev) [reply app/javascript/App/bundles/NewsfeedPage/NewsList.tsx]: It was worked as `toggle` but used as `close`. I've changed the initial function and renamed all to `closeTips`

File: app/javascript/App/bundles/NewsfeedPage/NewsfeedPage.tsx (hunk 4 above)
Comment (ashgaliyev) [reply app/javascript/App/bundles/NewsfeedPage/NewsfeedPage.tsx]: I didn't like that it has a `LOADING` state from the start. But currently, it's working precisely that way. So, I changed it by your suggestion

File: app/javascript/App/bundles/NewsfeedPage/NewsfeedPage.tsx (hunk 5 above)
Comment (ashgaliyev) [reply app/javascript/App/bundles/NewsfeedPage/NewsfeedPage.tsx]: Done.

File: app/javascript/App/bundles/NewsfeedPage/NewsfeedPage.tsx (hunk 6 above)
Comment (ashgaliyev) [reply app/javascript/App/bundles/NewsfeedPage/NewsfeedPage.tsx]: Done.

**Comment by hanchang** [comment]
Thanks for recording the remaining TODOs here @ashgaliyev -
https://trello.com/c/M1fQG8Qf/17-implement-changes-by-the-review

I'm going to go ahead and merge this and we can circle back around to those non-critical edits later.

**Pull request merged**